src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [ContentType(RazorLSPContentTypeDefinition.Name)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    [Export(typeof(ITextViewConnectionListener))]
```

Comment:
>ITextViewConnectionListener [](start = 19, length = 27)

Seems a bit unfortunate that we use this to track document lifetime via open views. Would it be possible to instead use an ITextDocument for lifetime?

Reply:
I'm definitely open to change the lifecycle tracking here. I'm unfamiliar how to do that with `ITextDocument`'s though. How would one do that?

Reply:
I *think* the text document should get created for you in VS. ITextDocumentFactoryService has events to listen for documents getting created/disposed. There is also an ITextBufferFactory.TextBufferCreated, but unfortunately no MEF export similar to ITextViewCreationListener for buffers or documents (I've pinged dpugh about this)

---
In reply to: [381463491](https://github.com/dotnet/aspnetcore-tooling/pull/1602#discussion_r381463491) [](ancestors = 381463491)

Reply:
Man that's awesome! Just tried it out but it looks like we're not overriding our content type soon enough:
![image](https://user-images.githubusercontent.com/2008729/74869174-5b9dc280-530c-11ea-8b2a-efa7d7bae3b8.png)

Filed an issue to move all this to that infrastructure: https://github.com/dotnet/aspnetcore/issues/19160

Reply:
Hmm, was playing around with setting our content type even earlier via RDT and still can't seem to set the content type before the text document is created.